yarn-project/sequencer-client/src/publisher/l1-publisher.ts

```diff
@@ -1042,26 +1047,54 @@ export class L1Publisher {
   private async sendProposeTx(
     encodedData: L1ProcessArgs,
     opts: { txTimeoutAt?: Date } = {},
+    timestamp: bigint,
   ): Promise<L1ProcessReturnType | undefined> {
     if (this.interrupted) {
       return undefined;
     }
     try {
       const kzg = Blob.getViemKzgInstance();
-      const { args, gas } = await this.prepareProposeTx(encodedData);
+      const { args, blobEvaluationGas } = await this.prepareProposeTx(encodedData);
       const data = encodeFunctionData({
         abi: this.rollupContract.abi,
         functionName: 'propose',
         args,
       });
+      let simulationResult = await this.l1TxUtils.simulateGasUsed(
+        {
+          to: this.rollupContract.address,
+          data,
+        },
+        {
+          time: timestamp + 1n,
+        },
+        [
+          {
+            address: this.rollupContract.address,
+            // @note we override checkBlob to false since blobs are not part simulate()
+            stateDiff: [
+              {
+                slot: toHex(9n, true),
```

Comment:
It'd be nice if we tweaked `generate-artifacts.sh` in `yarn-project` so we could extract the storage locations from the artifact output. If that ends up being too much, do you think we could add a unit test or something to check that this slot doesn't change?

Reply:
I think we could probably just use the `forge inspect <CONTRACT> storageLayout --json` to populate, then it should be fairly easy to fetch from there. Sounds like a good idea to have.

Its a kinda annoying one though, as ideally the variable itself should not exists 😬.